server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -494,6 +503,10 @@ public boolean isReadOnlyReplica() {
         return indexSettings.isSegRepEnabledOrRemoteNode() && isReadOnlyReplica;
     }
 
+    public boolean isUseCompoundFile() {
```

Comment:
Naming nitpick, but I'd prefer the method name `useCompoundFile()` as it reads more naturally.